pkgs/development/haskell-modules/configuration-common.nix

```
@@ -2151,6 +2151,14 @@ self: super: {
   # Restrictive bounds on aeson
   matrix-client = doJailbreak super.matrix-client;
 
+  cabal2nix = overrideCabal (old: {
+    postInstall = ''
+      ${old.postInstall or ""}
+      remove-references-to -t ${self.hpack} "$out/bin/cabal2nix"
+      remove-references-to -t ${self.distribution-nixpkgs} "$out/bin/hackage2nix"
```

I think it's nice to be explicit here, maybe even add a comment. The `data` output is absolutely required at runtime by hackage2nix.

```suggestion
      remove-references-to -t ${self.distribution-nixpkgs.out} "$out/bin/hackage2nix"
```